internal/registry/clientsecretrequest/rest.go

```diff
@@ -70,32 +125,159 @@ func (*REST) Categories() []string {
 }
 
 func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	t := trace.FromContext(ctx).Nest("create", trace.Field{
-		Key:   "kind",
-		Value: "OIDCClientSecretRequest",
-	})
+	t := trace.FromContext(ctx).Nest("create",
+		trace.Field{
+			Key:   "kind",
+			Value: "OIDCClientSecretRequest",
+		},
+		trace.Field{
+			Key:   "metadata.name",
+			Value: name(obj),
+		},
+	)
 	defer t.Log()
 
-	_, err := validateRequest(obj, t)
+	req, err := r.validateRequest(ctx, obj, createValidation, options, t)
 	if err != nil {
 		return nil, err
 	}
+	t.Step("validateRequest")
+
+	oidcClient, err := r.clients.Get(ctx, req.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			traceValidationFailure(t, fmt.Sprintf("client %q does not exist", req.Name))
+			errs := field.ErrorList{field.NotFound(field.NewPath("metadata", "name"), req.Name)}
+			return nil, apierrors.NewInvalid(kindFromContext(ctx), req.Name, errs)
+		}
+		traceFailureWithError(t, "clients.Get", err)
+		return nil, apierrors.NewInternalError(fmt.Errorf("getting client %q failed", req.Name))
+	}
+	t.Step("clients.Get")
+
+	rv, hashes, err := r.secretStorage.Get(ctx, oidcClient.UID)
+	if err != nil {
+		traceFailureWithError(t, "secretStorage.Get", err)
+		return nil, apierrors.NewInternalError(fmt.Errorf("getting secret for client %q failed", req.Name))
+	}
+	t.Step("secretStorage.Get")
+
+	var secret string
+	if req.Spec.GenerateNewSecret {
+		secret, err = generateSecret(r.rand)
+		if err != nil {
+			traceFailureWithError(t, "generateSecret", err)
+			return nil, apierrors.NewInternalError(fmt.Errorf("client secret generation failed"))
+		}
+		t.Step("generateSecret")
+
+		hash, err := bcrypt.GenerateFromPassword([]byte(secret), cost)
```

Comment:
This `generateFromPassword` call is the most likely to take a long time.
The `release note` should include notes about `Supervisor` requiring more `CPU` given this call.

Reply:
TODO: `fosite` contains some logic for `bcrypt` with tracing. Consider looking into this as a way to include logs around how long bcrypt operations are taking.  This could make it clear to an admin that they have or have not given supervisor enough CPU.